micro/exts_test.go

```diff
@@ -3,30 +3,23 @@ package micro
 import (
 	"testing"
 
-	"github.com/awalterschulze/gominikanren/sexpr"
 	"github.com/awalterschulze/gominikanren/sexpr/ast"
 )
 
 func TestOccurs(t *testing.T) {
-	tests := []func() (string, string, Substitutions, bool){
-		deriveTupleO(",x", ",x", Substitutions(nil), true),
-		deriveTupleO(",x", ",y", nil, false),
-		deriveTupleO(",x", "(,y)", Substitutions{
-			"y": ast.NewVariable("x"),
+	x := ast.NewVar("x", 0)
+	y := ast.NewVar("y", 1)
+	tests := []func() (*ast.SExpr, *ast.SExpr, Substitutions, bool){
+		deriveTuple3SVars(x, x, Substitutions(nil), true),
+		deriveTuple3SVars(x, y, nil, false),
+		deriveTuple3SVars(x, ast.NewList(y), Substitutions{
+			1: x,
```

Comment:
This is harder to read now, right?
What about:
```
Substitutions{
   y.Id(): x,
}
```

And same for all the substitutions below